tests/ui/associated-types/issue-23595-2.stderr

```diff
@@ -2,7 +2,13 @@ error[E0220]: associated type `anything_here_kills_it` not found for `Self`
   --> $DIR/issue-23595-2.rs:6:22
    |
 LL |     type B = C<Self::anything_here_kills_it>;
-   |                      ^^^^^^^^^^^^^^^^^^^^^^ help: `Self` has the following associated type: `B`
+   |                      ^^^^^^^^^^^^^^^^^^^^^^
+   |
+help: `Self` has the following associated type
+   |
+LL -     type B = C<Self::anything_here_kills_it>;
+LL +     type B = C<Self::B>;
+   |
```

Comment:
This suggestion is incorrect, we shouldn't suggest using the same associated type we are currently defining.